packages/client/src/layers/phaser/systems/createMapSystem.ts

```diff
@@ -54,21 +57,26 @@ export function createMapSystem(context: PhaserLayer, network: NetworkLayer) {
    * and draw it on the phaser tilemap
    */
   defineRxSystem(world, newTile$, async ({ x, y: z }) => {
-    const highestTiles = getHighestTilesAt({ x, z, perlin, Position, Position2D, Item });
-    if (!highestTiles) return;
+    requestIdleCallback(
+      () => {
+        const highestTiles = getHighestTilesAt({ x, z, perlin, Position, Position2D, Item });
+        if (!highestTiles) return;
 
-    const { y, foregroundTile, backgroundTile } = highestTiles;
+        const { y, foregroundTile, backgroundTile } = highestTiles;
 
-    const heightTile = HeightMapTiles[Math.max(-8, Math.min(8, Math.floor(y / 8)))];
-    if (heightTile != null) {
-      drawTile(x, z, heightTile, maps.height);
-    }
-    if (foregroundTile != null) {
-      drawTile(x, z, foregroundTile, maps.terrain, "Foreground");
-    }
-    if (backgroundTile != null) {
-      drawTile(x, z, backgroundTile, maps.terrain, "Background");
-    }
+        const heightTile = HeightMapTiles[Math.max(-8, Math.min(8, Math.floor(y / 8)))];
+        if (heightTile != null) {
+          drawTile(x, z, heightTile, maps.height);
+        }
+        if (foregroundTile != null) {
+          drawTile(x, z, foregroundTile, maps.terrain, "Foreground");
+        }
+        if (backgroundTile != null) {
+          drawTile(x, z, backgroundTile, maps.terrain, "Background");
+        }
+      },
+      { timeout: 3000 }
```

Comment:
Did you play with this without a timeout?

Reply:
yes, without a timeout it gets stuck if the operations phaser is doing every frame (redrawing the map) take longer than 16ms, because in that case there is no idle time for the callback to be executed in. If we could prevent phaser from redrawing every frame we could remove the timeout, but unfortunately I wasn't able to